### PR TITLE
[Step 5 Upgrade] Bump requirement to php 8.2

### PR DESCRIPTION
**Pull request by samsonasik**
@lisachenko here the 5th step 👍 , bump to php 8.2

### CODE REVIEW

**Comment by samsonasik** [comment]
implemented 🎉 @lisachenko Ready to merge 👍

File: src/ReflectionParameter.php [hunk 1]
```diff
@@ -320,6 +320,10 @@ public function getType(): ?\ReflectionType
             return null;
         }
 
+        if ($parameterType === 'iterable') {
+            $parameterType = 'Traversable|array';
+        }
```
Comment (samsonasik) [inline src/ReflectionParameter.php:325]: @lisachenko this is needed as php 8.2 display `Traversable|array` for `iterable, see https://3v4l.org/3uVi3

File: src/Traits/ReflectionFunctionLikeTrait.php [hunk 2]
```diff
@@ -198,6 +198,10 @@ public function getReturnType(): ?\ReflectionType
             return null;
         }
 
+        if ($returnType === 'iterable') {
+            $returnType = 'Traversable|array';
+        }
```
Comment (samsonasik) [inline src/Traits/ReflectionFunctionLikeTrait.php:203]: @lisachenko this is needed as php 8.2 display `Traversable|array` for `iterable, see https://3v4l.org/V1RCP

**Comment by TomasVotruba** [comment]
Seems good to go :+1:

File: src/ReflectionParameter.php (hunk 1 above)
Comment (lisachenko) [reply src/ReflectionParameter.php:325]: Oh, nice )) Haven't been aware about this 👍

**Comment by lisachenko** [comment]
Here we go! 🕺

**Review by lisachenko — APPROVED** [review]

**Comment by lisachenko** [comment]
Btw, do you know where this annoying travis-ci check comes from? I've checked every tab, removed travis-ci from everywhere, but still this check is here...

**Comment by lisachenko** [comment]
Congrats, @samsonasik! 

The only missing final part here, for parser-reflection now is ReflectionAttribute + relevant attributes API, which can be ported from https://github.com/goaop/parser-reflection/pull/122/files#diff-538c6d3a73a6c88b7717c7483cefab2a2f7c9ff32c5267a96ed2d6cbaaccc0d3

**Comment by samsonasik** [comment]
@lisachenko thank you for merging, I will look into the attribute part. I don't know about travis CI part as I don't have admin access on this repo, probably activated in API request somewhere in `/settings/installations` `settings/hooks` need to be disabled.

**Pull request merged**